README.md

```diff
@@ -11,51 +11,61 @@ related-issues: (will contain links to implementation PRs)
 # Summary
 [summary]: #summary
 
-Make most attribute definitions in `pkgs/top-level/all-packages.nix` be auto-generated from a single flat directory instead, where the subdirectory corresponds to the attribute name. The ad-hoc category-based structure of packages will be gotten rid of.
+Make trivial top-level attribute definitions in `pkgs/top-level/all-packages.nix` be auto-generated from a predictable attribute-based file hierarchy.
+This makes it much easier to contribute new packages packages, since there's no more guessing needed as to where the package should go, both in the ad-hoc directory categories and in `pkgs/top-level/all-packages.nix`.
+
 
 # Motivation
 [motivation]: #motivation
 
-- (Especially new) package contributors are having a hard time figuring out which files to add and edit. These are very common questions:
-  - Which directory should my package definition go in? What are all the categories and do they matter? What if the package has multiple matching categories?
-  - Why can't I build my package after adding the package file? [introduced to all-packages.nix] Where in all-packages.nix should my package go?
-- Figuring out where an attribute is defined is very tricky:
+- (Especially new) package contributors are having a hard time figuring out which files to add or edit. These are very common questions:
```

Comment:
I am not sure if that will improve the situation for new people. They will still have questions and they will just be different.

Reply:
It will certainly improve the situation! It removes some of the most annoying problems to deal with when contributing changes to packages. And yes this won't fix all the problems, just one of the most prominent ones. And just because there's always more problems doesn't mean solving one won't improve the situation.